### ISSUE TITLE
Incorrect type definition for Rehydrated (includes fix PR)

### ISSUE THREAD

**Issue by tallpants** [awslabs/aws-mobile-appsync-sdk-js#205]
Hey guys!

So right now the `Rehydrated` element from `aws-appsync-react` just will not work in a TypeScript project. This is because it expects children to be of type `React.ReactChildren`.

If you just do:

```js
// ...
const App = () => <h1>Hello, World!</h1>;

const AppWithAppSyncProvider = () => (
  <ApolloProvider client={client}>
    <Rehydrated>
      <App />
    </Rehydrated>
  </ApolloProvider>
);
```

TypeScript will refuse to compile, and throws this error:

```
error TS2322: Type '{ children: Element; }' is not assignable to type 'Readonly<RehydratedProps>'.
  Types of property 'children' are incompatible.
    Type 'Element' is not assignable to type 'ReactChildren'.
      Property 'map' is missing in type 'Element'.

       <Rehydrated>
        ~~~~~~~~~~
```

This can be fixed by setting the type of the children expected by `Rehydrated` from `React.ReactChildren` to `React.ReactNode`.

I've tested this on my machine to make sure this works, and I'm opening a PR with the fix. I hope you folks can take a look at it and merge it in. Happy to answer any questions / make any changes you'd like.

PR here: #206

**Comment by tallpants** [comment]
@manueliglesias can you take a look at this please? It's a blocker for any TypeScript projects that use AppSync!

**Comment by tallpants** [comment]
Closed by #206.
